### PR TITLE
not removing trailing \n in contents when encoding, as it is probably…

### PR DESCRIPTION
**Pull request by crystina-z**
… the delimiter before an empty field; add field length checking

To solve the issue that @NThakur20 met when encoding SciFact documents

### CODE REVIEW

**Comment by lintool** [comment]
@crystina-z can we write a test case?

**Comment by NThakur20** [comment]
Tried out locally, works for SCIDOCS and FiQA.
This helps to encode datasets such as SCIDOCS or FIQA, where the title might be present but the text empty. 

Kind Regards,
Nandan Thakur

File: pyserini/encode/_base.py [hunk 1]
```diff
@@ -88,6 +88,25 @@ def __iter__(self):
                 to_yield[key] = self.all_info[key][idx: min(idx + self.batch_size, end_idx)]
             yield to_yield
 
+    def _parse_fields_from_contents(self, contents):
```
Comment (crystina-z) [inline pyserini/encode/_base.py:91]: adding a separate function to process the contents, deciding if we should remove the trailing \n. 
As it seems CACM has a trailing \n for all documents

**Comment by crystina-z** [comment]
and added the test case

**Comment by lintool** [comment]
hey @NThakur20 can you sign off on this?

**Review by NThakur20 — APPROVED** [review]
Looks good to me!

Kind Regards,
Nandan Thakur

**Comment by crystina-z** [comment]
`python -m unittest` passed

**Pull request merged**